kolena/workflow/visualization/_activation_map.py

```diff
@@ -0,0 +1,129 @@
+        return self._scale_to_colormap(intensity)
+
+    def blue(self, intensity: np.uint8) -> float:
+        return self._scale_to_colormap(intensity + MAX_UINT8 / 4)
+
+
+def colorize_activation_map(activation_map: np.ndarray, fade_low_activation: bool = True) -> np.ndarray:
+    """
+    Applies the [MATLAB "Jet" colormap](http://blogs.mathworks.com/images/loren/73/colormapManip_14.png) to the
+    activation map.
+
+    :param np.ndarray activation_map: A 2D numpy array, shaped (h, w) or (h, w, 1), of the activation map in `np.uint8`
+        or `float` ranging [0, 1].
+    :param bool fade_low_activation: Fades out the regions with low activation by applying zero alpha value if set
+        `True`; otherwise, activation map is shown as is without any fading applied. By default, it's set to `True`.
+        This option makes the overlay visualization better by highlighting only the important regions.
+    :return: np.ndarray colorized activation map in BGRA format, in (h, w, 4) shape.
+    """
+    if activation_map.size == 0:
+        raise InputValidationError("input array is empty")
+
+    if np.issubdtype(activation_map.dtype, np.floating) and ((0 <= activation_map) & (activation_map <= 1)).all():
+        activation_map = np.rint(activation_map * MAX_UINT8).astype(np.uint8)
+
+    if activation_map.dtype != np.uint8:
+        raise InputValidationError(f"input array type must be np.uint8, but received {activation_map.dtype}")
+
+    if len(activation_map.shape) != 2 and len(activation_map.shape) != 3:
+        raise InputValidationError(
+            f"input array must have 2 or 3 dimensions, but received {len(activation_map.shape)}\n"
+            "Expected shape is (h, w) or (h, w, 1)",
+        )
+
+    if len(activation_map.shape) == 2:
+        activation_map = np.expand_dims(activation_map, axis=2)
+
+    if len(activation_map.shape) == 3:
+        if activation_map.shape[2] != 1:
+            raise InputValidationError(
+                f"input image must be a single-channel, but received {activation_map.shape[2]} channels\n"
+                "Expected shape is (h, w) or (h, w, 1)",
+            )
+
+        activation_map = np.repeat(activation_map, 4, axis=2)
+        if fade_low_activation:
+            # apply non-linear scaling alpha channel [0, 255]
+            alpha = MAX_UINT8 / (1 + np.exp(((MAX_UINT8 / 2) - activation_map[:, :, 1]) / 8))
+            activation_map[:, :, 3] = alpha
+
+    colormap = ColorMapJet()
+    for i, row in enumerate(activation_map):
+        for j, pixel in enumerate(row):
+            intensity = pixel[0]
+            activation_map[i][j] = np.array(
+                [
+                    colormap.blue(intensity),
+                    colormap.green(intensity),
+                    colormap.red(intensity),
+                    pixel[3] if fade_low_activation else MAX_UINT8,
+                ],
+                dtype=np.uint8,
+            )
```

Comment:
Is there any way for us to apply this as a vectorized function? I would expect nested `for` loops to be quite slow.